openmc/plotter.py

```diff
@@ -0,0 +1,910 @@
+from numbers import Integral, Real
+from itertools import chain
+import string
+
+import numpy as np
+
+import openmc.checkvalue as cv
+import openmc.data
+
+# Supported keywords for continuous-energy cross section plotting
+PLOT_TYPES = ['total', 'scatter', 'elastic', 'inelastic', 'fission',
+              'absorption', 'capture', 'nu-fission', 'nu-scatter', 'unity',
+              'slowing-down power', 'damage']
+
+# Supported keywoards for multi-group cross section plotting
+PLOT_TYPES_MGXS = ['total', 'absorption', 'scatter', 'fission',
+                   'kappa-fission', 'nu-fission', 'prompt-nu-fission',
+                   'deleyed-nu-fission', 'chi', 'chi-prompt', 'chi-delayed',
+                   'inverse-velocity', 'beta', 'decay rate', 'unity']
+# Create a dictionary which can be used to convert PLOT_TYPES_MGXS to the
+# openmc.XSdata attribute name needed to access the data
+_PLOT_MGXS_ATTR = {line: line.replace(' ', '_').replace('-', '_')
+                   for line in PLOT_TYPES_MGXS}
+_PLOT_MGXS_ATTR['scatter'] = 'scatter_matrix'
+
+# Special MT values
+UNITY_MT = -1
+XI_MT = -2
+
+# MTs to combine to generate associated plot_types
+_INELASTIC = [mt for mt in openmc.data.SUM_RULES[3] if mt != 27]
+PLOT_TYPES_MT = {'total': openmc.data.SUM_RULES[1],
+                 'scatter': [2] + _INELASTIC,
+                 'elastic': [2],
+                 'inelastic': _INELASTIC,
+                 'fission': [18],
+                 'absorption': [27], 'capture': [101],
+                 'nu-fission': [18],
+                 'nu-scatter': [2] + _INELASTIC,
+                 'unity': [UNITY_MT],
+                 'slowing-down power': [2] + _INELASTIC + [XI_MT],
+                 'damage': [444]}
+# Operations to use when combining MTs the first np.add is used in reference
+# to zero
+PLOT_TYPES_OP = {'total': (np.add,),
+                 'scatter': (np.add,) * (len(PLOT_TYPES_MT['scatter']) - 1),
+                 'elastic': (),
+                 'inelastic': (np.add,) * (len(PLOT_TYPES_MT['inelastic']) - 1),
+                 'fission': (), 'absorption': (),
```

Comment:
Nit: Same as on line 37, be more readable to keep map construction consistent here by keeping one element per line.